### ISSUE TITLE
Updating from 3.3.12 to 4.0.0 breaks our production deployment

### ISSUE THREAD

**Issue by Glinkis** [webpack/webpack-cli#1970]
After upgrading from `3.3.12` to `4.0.0`, we get **404** for our index.html page.
We're using [serve](https://www.npmjs.com/package/serve) inside a docker container to serve our app in production.
If I do a local build, and run `serve`, everything works as expected.

Our dockerfile looks like this:
```dockerfile
FROM node:12.16

# Copy only the files relevant to the installation
COPY ./yarn.lock /
COPY ./package.json /
COPY ./packages/babel/package.json /packages/babel/package.json
COPY ./packages/webapp/package.json /packages/webapp/package.json
COPY ./packages/shared/package.json /packages/shared/package.json

# Install dependencies recursively for all packages
RUN yarn install --frozen-lockfile

# Copy the rest of the directory
COPY ./ /

ARG NODE_ENV
ENV NODE_ENV=$NODE_ENV
ARG API_URL
ENV API_URL=$API_URL

WORKDIR /packages/webapp

RUN export NODE_ENV=$NODE_ENV && export API_URL=$API_URL && yarn build

EXPOSE 8080

CMD ["yarn", "serve"]
```

The `"serve"` script is `"serve": "serve -l 8080 -c ../serve.json",`

And the serve config like this:

```json
{
  "public": "dist",
  "directoryListing": false,
  "rewrites": [
    {
      "source": "/static/**",
      "destination": "/static"
    },
    {
      "source": "/**",
      "destination": "/index.html"
    }
  ]
}
```

**Comment by evilebottnawi** [comment]
Sorry don't know, I don't know what your configuration mean, for running serve you need write `webpack serve`, we never support `json` config in this format

**Comment by Glinkis** [comment]
I wrote that we're using the `serve` package, not webpack serve. If you're going to answer, please read the full description.

**Comment by evilebottnawi** [comment]
Sorry, `serve` is not related to webpack-cli, so I don't know why it does not work, if you want to help, please use the issue template and describe your problems,  `Updating from 3.3.12 to 4.0.0 breaks our production deployment` is not help to solve your problem

**Comment by Glinkis** [comment]
I thought I described my problem quite thoroughly. I'm not sure what else to add?

**Comment by evilebottnawi** [comment]
@Glinkis please create minimum reproducible test repo, I don't know what your Docker file do

**Comment by Glinkis** [comment]
It seems my issue was that the `--env` flag was changed in version 4. So I fixed it by following the advice in #1216
